src/Compiler/pars.fsy

```diff
@@ -1822,32 +1822,65 @@ classDefnMembersAtLeastOne:
 
 /* The "with get, set" part of a member definition */
 classDefnMemberGetSet: 
-  | WITH classDefnMemberGetSetElements
+  | WITH classDefnMemberGetSetElement AND classDefnMemberGetSetElement
+     { let mWithKwd = rhs parseState 1
+       let mAndKwd = rhs parseState 3
+       mWithKwd, $2, mAndKwd, $4 }
+
+  | OWITH classDefnMemberGetSetElement AND classDefnMemberGetSetElement OEND
+     { let mWithKwd = rhs parseState 1
+       let mAndKwd = rhs parseState 3
+       mWithKwd, $2, mAndKwd, $4 }
+
+  | OWITH classDefnMemberGetSetElement AND classDefnMemberGetSetElement error
+     { let mWithKwd = rhs parseState 1
+       let mAndKwd = rhs parseState 3
+       reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedWith())
+       mWithKwd, $2, mAndKwd, $4 }
+
+/* The "with get" part of a member definition */
+classDefnMemberGet:
+  | WITH classDefnMemberGetSetElement
      { let mWithKwd = rhs parseState 1
        mWithKwd, $2  }
 
-  | OWITH classDefnMemberGetSetElements OEND
+  | OWITH classDefnMemberGetSetElement OEND
      { let mWithKwd = rhs parseState 1
        mWithKwd, $2  }
 
-  | OWITH classDefnMemberGetSetElements error
+  | OWITH classDefnMemberGetSetElement error
      { let mWithKwd = rhs parseState 1
        reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedWith())
        mWithKwd, $2  }
 
-/* The "get, set" part of a member definition */
-classDefnMemberGetSetElements: 
-  | classDefnMemberGetSetElement 
-     { [$1], None  }
-  | classDefnMemberGetSetElement AND classDefnMemberGetSetElement
-     { let mAnd = rhs parseState 2
-       [$1;$3], Some mAnd }
+classDefnMemberGetSetPattern:
+  | LPAREN rparen
+     { let m = rhs2 parseState 1 2
+       SynPat.Const(SynConst.Unit, m) }
+
+  | atomicPattern 
+     { $1 }
 
 classDefnMemberGetSetElement: 
-  | opt_inline opt_attributes bindingPattern opt_topReturnTypeWithTypeConstraints EQUALS typedSequentialExprBlock 
-     { let mEquals = rhs parseState 5
-       let mRhs = ($6 : SynExpr).Range 
-       ($1, $2, $3, $4, Some mEquals, $6, mRhs) }
+  | opt_inline opt_attributes ident classDefnMemberGetSetPattern opt_topReturnTypeWithTypeConstraints EQUALS typedSequentialExprBlock
+     { let mSet = rhs parseState 3
+       let getOrSet = $3.idText
+       printfn "getOrSet %s" getOrSet
+       if getOrSet <> "get" && getOrSet <> "set" then raiseParseErrorAt mSet (FSComp.SR.parsGetAndOrSetRequired()) else
+
+       let attrs = $2 |> List.map (fun attrList -> { attrList with Attributes = attrList.Attributes |> List.map (fun a -> { a with AppliesToGetterAndSetter = true } ) })
+       let mSet = rhs parseState 3
+       let mEquals = rhs parseState 6
+       let m = rhs2 parseState 1 7
```

Comment:
This range may include subsequent nodes when error happens inside `typedSequentialExprBlock`. A safer approach would be to get the range of the parsed expression and to join it with the start range. Here's an example of a similar range issue related to error recovery: https://github.com/dotnet/fsharp/pull/12224